hibernate5/src/test/java/com/baeldung/hibernate/proxy/HibernateProxyUnitTest.java

```diff
@@ -0,0 +1,75 @@
+package com.baeldung.hibernate.proxy;
+
+import org.hibernate.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+public class HibernateProxyUnitTest {
+
+    private Session session;
+
+    @Before
+    public void init(){
+        try {
+            session = HibernateUtil.getSessionFactory("hibernate.properties")
+                    .openSession();
+        } catch (HibernateException | IOException e) {
+            fail("Failed to initiate Hibernate Session [Exception:" + e.toString() + "]");
+        }
+
+        Boss boss = new Boss("Eduard", "Freud");
+        session.save(boss);
+    }
+
+    @After
+    public void close(){
+        if(session != null) {
+            session.close();
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void givenAnInexistentEmployeeId_whenUseGetMethod_thenReturnNull() {
+        Employee employee = session.get(Employee.class, new Long(14));
+        assertNull(employee);
+        employee.getId();
```

Comment:
I don't understand the purpose of this line. Didn't we just assert that employee is null? So why are we calling it?